src/modules/UI/Wallets/action.js

```diff
@@ -150,15 +150,13 @@ export const setEnabledTokens = (walletId: string, enabledTokens: Array<string>,
   // get a copy of the relevant core wallet
   const wallet = CORE_SELECTORS.getWallet(state, walletId)
   // now actually tell the wallet to enable the token(s) in the core and save to file
-  WALLET_API.setEnabledTokens(wallet, enabledTokens, disabledTokens)
-    .then(() => {
-      // let Redux know it was completed successfully
-      dispatch(setTokensSuccess())
-      dispatch(updateWalletEnabledTokens(walletId, enabledTokens))
-      // refresh the wallet in Redux
-      dispatch(refreshWallet(walletId))
-    })
-    .catch(e => dispatch(displayErrorAlert(e.message)))
+  return WALLET_API.setEnabledTokens(wallet, enabledTokens, disabledTokens).then(() => {
+    // let Redux know it was completed successfully
```

Comment:
Hypothetically we could consolidate this into one atomic Redux action. Then again I know you guys were debating that pattern the other day so you're probably aware of it.